daffodil-runtime1/src/main/scala/org/apache/daffodil/debugger/InteractiveDebugger.scala

```diff
@@ -1720,7 +1722,7 @@ class InteractiveDebugger(runner: InteractiveDebuggerRunner, eCompilers: Express
                         |Immediately abort all processing.""".stripMargin
       override lazy val short = "q"
       def act(args: Seq[String], state: ParseOrUnparseState, processor: Processor): DebugState.Type = {
-        sys.exit(1)
+        throw new DebuggerExitException("")
```

Comment:
If the string is always going to be the empty string, I suggest just removing the `str: String` parameter.

Reply:
Also, this exception isn't caught in the CLI, so if this is thrown it's going to show up as a bug. The CLI should catch this and output an appropriate error code. Probably ExitCode.Success is reasonable. since the debugger exiting is probably not considered a failure.